src/platform/plugins/shared/workflows_management/public/widgets/workflow_yaml_editor/lib/store/types.ts

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+import type { WorkflowGraph } from '@kbn/workflows/graph';
+import type YAML from 'yaml';
+import type { WorkflowLookup } from './utils/build_workflow_lookup';
+
+// State interface - only serializable data
+export interface WorkflowEditorState {
+  yamlString?: string;
+  computed?: {
+    yamlDocument?: YAML.Document; // This will be handled specially for serialization
+    workflowLookup?: WorkflowLookup;
+    workflowGraph?: WorkflowGraph; // This will be handled specially for serialization
+  };
+  focusedStepId?: string;
+}
+
+// Store types (will be properly typed when store.ts is imported)
+export type WorkflowEditorStore = ReturnType<typeof import('./store').createWorkflowEditorStore>;
+export type RootState = ReturnType<WorkflowEditorStore['getState']>;
```

Comment:
And define the state shape explicitly, so it can be enforced in the store creation.

```suggestion
// Root state type
export interface RootState {
  workflow: WorkflowEditorState;
}
export type WorkflowEditorStore = EnhancedStore<RootState>;
```